frontend/lib/activation/deeplink_activation.dart

```diff
@@ -189,15 +199,20 @@ class _WarningText extends StatelessWidget {
   }
 }
 
-DynamicActivationCode? getActivationCode(BuildContext context, String base64qrcode) {
+DynamicActivationCode? getActivationCode(
+    BuildContext context, String base64qrcode, Function(String message) updateErrorMessage) {
   try {
     final activationCode = const ActivationCodeParser().parseQrCodeContent(const Base64Decoder().convert(base64qrcode));
     return activationCode;
+  } on CardExpiredException catch (e, _) {
+    updateErrorMessage(
+        t.deeplinkActivation.cardExpiredExceptionMessage(expiryDate: DateFormat('dd.MM.yyyy').format(e.expiry)));
```

Comment:
:thought_balloon: We should avoid to use hardcoded date formats like this. However, since we do this in a lot of places already, I created a separate [issue](https://github.com/digitalfabrik/entitlementcard/issues/1942) so no need to change anything here.